node/nodeTransactions_test.go

```diff
@@ -102,13 +118,166 @@ func TestNode_GetTransaction_ShouldFindInTxStorageAndReturn(t *testing.T) {
+	}
+	storer := &mock.ChainStorerMock{
+		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+			return getStorerStub(true)
+		},
+	}
+
+	blockHash := []byte("hash")
+	mbHash := []byte("mbHash")
+	epoch := uint32(10)
+	sndShard := uint32(1)
+	rcvShard := uint32(2)
+	round := uint64(123)
+	blockNonce := uint64(1001)
+	n, _ := node.NewNode(
+		node.WithDataPool(dataPool),
+		node.WithDataStore(storer),
+		node.WithInternalMarshalizer(&mock.MarshalizerFake{}, 0),
+		node.WithAddressPubkeyConverter(&mock.PubkeyConverterMock{}),
+		node.WithShardCoordinator(&mock.ShardCoordinatorMock{}),
+		node.WithHistoryProcessor(&mock.HistoryProcessorStub{
+			IsEnabledCalled: func() bool {
+				return true
+			},
+			GetTransactionCalled: func(hash []byte) (*fullHistory.HistoryTransactionWithEpoch, error) {
+				return &fullHistory.HistoryTransactionWithEpoch{
+					Epoch: epoch,
+					HistoryTransaction: &fullHistory.HistoryTransaction{
+						MbHash:      mbHash,
+						HeaderHash:  blockHash,
+						HeaderNonce: blockNonce,
+						SndShardID:  sndShard,
+						RcvShardID:  rcvShard,
+						Round:       round,
+					},
+				}, nil
+			},
+		}),
+	)
+
+	dummyTx, _ := getDummyNormalTx()
+	expectedTx := &transaction.ApiTransactionResult{
+		Type:       "normal",
+		Nonce:      dummyTx.Nonce,
+		Round:      round,
+		Epoch:      epoch,
+		Value:      dummyTx.Value.String(),
+		Receiver:   hex.EncodeToString(dummyTx.RcvAddr),
+		Sender:     hex.EncodeToString(dummyTx.SndAddr),
+		GasPrice:   dummyTx.GasPrice,
+		GasLimit:   dummyTx.GasLimit,
+		Data:       string(dummyTx.Data),
+		Code:       "",
+		Signature:  hex.EncodeToString(dummyTx.Signature),
+		SndShard:   sndShard,
+		RcvShard:   rcvShard,
+		BlockNonce: blockNonce,
+		MBHash:     hex.EncodeToString(mbHash),
+		BlockHash:  hex.EncodeToString(blockHash),
+		Status:     core.TxStatusExecuted,
+	}
+
+	tx, err := n.GetTransaction("aaaa")
+	assert.NoError(t, err)
+	assert.Equal(t, expectedTx, tx)
+}
+
+func TestNode_GetFullHistoryTransaction_TxInPoolShouldReturnItDirectly(t *testing.T) {
+	t.Parallel()
+
+	dataPool := &testscommon.PoolsHolderStub{
+		TransactionsCalled:         getCacherHandler(true, ""),
+		RewardTransactionsCalled:   getCacherHandler(false, ""),
+		UnsignedTransactionsCalled: getCacherHandler(false, ""),
+	}
+	storer := &mock.ChainStorerMock{
+		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+			return getStorerStub(true)
+		},
+	}
+
+	n, _ := node.NewNode(
+		node.WithDataPool(dataPool),
+		node.WithDataStore(storer),
+		node.WithInternalMarshalizer(&mock.MarshalizerFake{}, 0),
+		node.WithAddressPubkeyConverter(&mock.PubkeyConverterMock{}),
+		node.WithShardCoordinator(&mock.ShardCoordinatorMock{}),
+		node.WithHistoryProcessor(&mock.HistoryProcessorStub{
+			IsEnabledCalled: func() bool {
+				return true
+			},
+		}),
+	)
+
+	dummyTx, _ := getDummyNormalTx()
+	expectedTx := &transaction.ApiTransactionResult{
```

Comment:
Test is not really obvious (not visible where `expectedTx` is added in the pool).

Reply:
TransactionsCalled:         getCacherHandler(true, "") yeah is not obvious because getCacherHandler method will return a cacher with a transaction that is created with method getDummyNormalTx()